### PR TITLE
Compute RH, avoid re-computations

### PR DESCRIPTION
**Pull request by charleskawczynski**
`aux_gm.T` and `aux_gm.buoy` are already computed above in this function, so this PR removes the re-computations to avoid potential inconsistencies. This PR also adds computations for `aux_gm.RH`, which were previously computed once during initialization.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
This PR is quite interesting

File: src/update_aux.jl [hunk 1]
```diff
@@ -84,14 +84,19 @@ function update_aux!(edmf, gm, grid, state, Case, param_set, TS)
...
-    #! format: off
     get_GMV_CoVar(edmf, grid, state, :Hvar, :θ_liq_ice)
     get_GMV_CoVar(edmf, grid, state, :QTvar, :q_tot)
     get_GMV_CoVar(edmf, grid, state, :HQTcov, :θ_liq_ice, :q_tot)
     GMV_third_m(edmf, grid, state, :Hvar, :θ_liq_ice, :H_third_m)
     GMV_third_m(edmf, grid, state, :QTvar, :q_tot, :QT_third_m)
     GMV_third_m(edmf, grid, state, :tke, :w, :W_third_m)
-    #! format: on
+
+    @inbounds for k in real_center_indices(grid)
```
Comment (ilopezgp) [inline src/update_aux.jl:94]: Can't this be added to the loop in line 80?

File: src/update_aux.jl [hunk 2]
```diff
@@ -222,10 +227,10 @@ function update_aux!(edmf, gm, grid, state, Case, param_set, TS)
     #####
     a_up_bulk = aux_tc.bulk.area
     @inbounds for k in real_center_indices(grid)
+        # TODO: Can we use thermo states here? Is this consistent with one constructed
+        #       from a thermo state?
         aux_gm.q_liq[k] = (a_up_bulk[k] * aux_tc.bulk.q_liq[k] + (1 - a_up_bulk[k]) * aux_en.q_liq[k])
         aux_gm.q_ice[k] = (a_up_bulk[k] * aux_tc.bulk.q_ice[k] + (1 - a_up_bulk[k]) * aux_en.q_ice[k])
-        aux_gm.T[k] = (a_up_bulk[k] * aux_tc.bulk.T[k] + (1 - a_up_bulk[k]) * aux_en.T[k])
```
Comment (ilopezgp) [inline src/update_aux.jl:227]: I don't see the update to `aux_gm.T[k]` anywhere before this in this file.

File: src/update_aux.jl [hunk 3]
```diff
@@ -222,10 +227,10 @@ function update_aux!(edmf, gm, grid, state, Case, param_set, TS)
...
-        aux_gm.buoy[k] = (a_up_bulk[k] * aux_tc.bulk.buoy[k] + (1 - a_up_bulk[k]) * aux_en.buoy[k])
```
Comment (ilopezgp) [inline src/update_aux.jl:228]: I actually think we have to keep this. If you see where it is defined above, we are computing the GMV buoyancy to subtract it to updrafts and environment. Therefore, we have to recompute it here -- it should be equal to zero in this second step.

Comment (ilopezgp) [reply src/update_aux.jl:228]: This is due to the implied balance in the SCM, see equation (47) in Cohen et al.

File: src/update_aux.jl (hunk 2 above)
Comment (charleskawczynski) [reply src/update_aux.jl:227]: Oh, yeah, maybe I mixed this up

**Comment by charleskawczynski** [comment]
I think #507 basically fixed things here. Closing.

**Pull request closed without merging**